### ISSUE TITLE
Update the auth permissions so that create repo requires admin permission

### ISSUE THREAD

**Issue by endophage** [docker/notary#845]
I think that means all permissions (*), not just push/pull.

Confirmation from @jlhawn that scope=repository:library/foobar:* is given only to namespace admins, as opposed to anyone with read-write access.

**Comment by cyli** [comment]
IRL discussion and update from https://github.com/docker/notary/issues/789#issuecomment-231880075:

1.  Have configuration about whether this permission is hard-enforced or soft-enforced (if hard-enforced, we don't accept a POST to the update endpoint if the repo doesn't already exist)
2. Add a new endpoint for creating a new repo that requires a `*` permission, and new clients when initializing a repo for the first time will use this new endpoint

**Comment by jlhawn** [comment]
@endophage @cyli yes, `*` access is given to admins of the repository namespace. You may also want to consider adding a new access scope for creating a repository. Something like `repo-namespace:library:create-repo` would only be given to accounts which are authorized to create new repositories in the `library` namespace. This would require a change in the authorization backend on the server issuing the tokens though.
